src/main/java/org/folio/service/impl/RequestServiceImpl.java

```diff
@@ -67,7 +67,13 @@ public RequestWrapper createSecondaryRequest(Request request, String borrowingTe
         return executionService.executeSystemUserScoped(lendingTenantId, () -> {
           log.info("createSecondaryRequest:: creating requester {} in lending tenant ({})",
             requesterId, lendingTenantId);
-          userCloningService.clone(primaryRequestRequester);
+          User shadowUser = userCloningService.clone(primaryRequestRequester);
+
+          if (primaryRequestRequester.getPatronGroup() != null &&
+            !primaryRequestRequester.getPatronGroup().equals(shadowUser.getPatronGroup())) {
+            shadowUser.setPatronGroup(primaryRequestRequester.getPatronGroup());
+            userService.update(shadowUser);
+          }
```

Comment:
- Please extract this into a separate method
- you are calling `primaryRequestRequester.getPatronGroup()` 3 times, should we extract it into a variable?